compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -257,6 +218,7 @@ class TypeApplications(val self: Type) extends AnyVal {
   def hkResult(implicit ctx: Context): Type = self.dealias match {
     case self: TypeRef => self.info.hkResult
     case self: RefinedType => NoType
+    case self: AppliedType => NoType
```

Comment:
This on the other hand would definitely be wrong for curried type lambdas :)

Reply:
Fixed :-)